### ISSUE TITLE
ParamSpec masks captured default values with ellipsis

### ISSUE THREAD

**Issue by ringohoffman** [microsoft/pyright#4265]
**Is your feature request related to a problem? Please describe.**

Here is a minimally reproducible example:

```python
from typing import Callable, Concatenate, ParamSpec, TypeVar


P = ParamSpec("P")
R = TypeVar("R", covariant=True)


def decorator(f: Callable[P, R]) -> Callable[Concatenate[int, P], R]:
    def new_func(i: int, *args: P.args, **kwargs: P.kwargs) -> R:
        return f(*args, **kwargs)
    return new_func


@decorator
def a(j: int, k: int = 1) -> float:
    ...

a()
```

<img width="248" alt="Screenshot 2022-12-01 at 4 30 25 PM" src="https://user-images.githubusercontent.com/27844407/205172863-9eb16029-3f1f-4435-872e-a73d8c505303.png">

Is there any reason that the default values couldn't be shown?

**Describe the solution you'd like**

I want the actual default values to be shown, like this:

```
(function) a: (int, j: int, k: int = 1) -> float
```

**Comment by erictraut** [comment]
The ParamSpec captures the parameter names, types, and whether or not there is a default value. It doesn't capture the actual default value. Values are not tracked in the type system. So it makes sense to represent the default value as `...`. You'll see this convention in most type stub files.

Another reason not to show the value here is that type checkers don't evaluate values, only types. The value expression in your example is a numeric literal, but any value expression for is supported here (e.g. `foo(1, 2, 3)`). To get the actual value, you would need to execute the code.

**Comment by erictraut** [comment]
Actually, on second thought, we do print the expression used for the default value when presenting the type of a function that hasn't been captured by a `ParamSpec`, so this is inconsistent. (Note that pyright doesn't evaluate the value in this case; it simply prints the expression as it was found. So if the expression is `1 + 2`, it will be presented as `1 + 2`, not as `3`.) Pyright could track the expression with the captured ParamSpec and output that same expression in the resulting type. That would eliminate the inconsistency between a decorated and undecorated function.

**Comment by erictraut** [comment]
This will be addressed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.283, which I just published. It will also be included in a future version of pylance.
